File: ios-app/IzmirEshot/Models/Stop.swift
import Foundation
import CoreLocation

struct Stop: Identifiable, Codable, Hashable {
    let id: String
    let name: String
    let latitude: Double
    let longitude: Double
    let lineIds: String
    
    var coordinate: CLLocationCoordinate2D {
        CLLocationCoordinate2D(latitude: latitude, longitude: longitude)
    }
}

File: android-app/app/src/main/java/com/eshot/app/data/model/Stop.kt
package com.eshot.app.data.model

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "stops")
data class Stop(
    @PrimaryKey val id: String,
    val name: String,
    val latitude: Double,
    val longitude: Double,
    val lineIds: String // Comma separated IDs or JSON array
)
